### PR TITLE
Add statsig example

### PR DESCRIPTION
**Pull request by goncy**
image

Updated Statsig example to use the REST Api instead of the SDK as its not currently compatible with our edge

### CODE REVIEW

**Review by gbibeaul — APPROVED** [review]

File: edge-functions/ab-testing-statsig/pages/[bucket].tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  // Return the response
+  return response
+}
+`}</Snippet>
+        <Text>
+          Once the page is fully functional, we have to log the exposure for the
+          experiment, this will let Statsig know that the bucket was correctly
+          assigned and the user has been exposed to the experiment. If we
+          don&apos;t log the exposure, all users will get the same bucket and
+          the traffic won&apos;t be split.
```
Comment (lfades) [inline edge-functions/ab-testing-statsig/pages/[bucket].tsx:129]: users will all get the same experiment if the traffic doesn't get reported to Statsig? why is that?

File: edge-functions/ab-testing-statsig/pages/[bucket].tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+}, [bucket])`}</Snippet>
+        <Text className="text-gray-500 border-l-4 pl-2">
+          Statsig REST Api calls were implemented in another file for sake of
+          simplicity and exported as <Code>logExposure</Code> and{' '}
+          <Code>getExperiment</Code> methods. You can find those{' '}
+          <Link href="https://github.com/vercel/examples/tree/main/edge-functions/ab-testing-statsig/api.ts">
+            here
+          </Link>
+          .
+        </Text>
```
Comment (lfades) [inline edge-functions/ab-testing-statsig/pages/[bucket].tsx:147]: I think we can remove this paragraph completely 👍

File: edge-functions/ab-testing-statsig/pages/[bucket].tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        <Text>
+          You can reset the bucket multiple times to get a different bucket
+          assigned. You can configure your experiments, see diagnostics and
+          results in your account.{' '}
+          <Link href="https://console.statsig.com/">Statsig console</Link>.
```
Comment (lfades) [inline edge-functions/ab-testing-statsig/pages/[bucket].tsx:152]: Links to external sites should be using `target="_blank"`

File: edge-functions/ab-testing-statsig/pages/[bucket].tsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    >
+      <Layout title="statsig-metric" path="solutions/statsig-metric">
+        <Component {...pageProps} />
+      </Layout>
+    </StatsigProvider>
+  )
+}
+`}</Snippet>
+        <Text>
+          After we can tracks events by calling using the{' '}
```
Comment (lfades) [inline edge-functions/ab-testing-statsig/pages/[bucket].tsx:196]:
```suggestion
          Now we can tracks events by calling using the{' '}
```

File: edge-functions/ab-testing-statsig/pages/[bucket].tsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+  function resetBucket() {
+    Cookie.remove('uid')
+    Statsig.logEvent('reset-bucket')
+    reload()
+  }
+
+  useEffect(() => {
+    // Log exposure to statsig
+    api.logExposure(Cookie.get('uid')!, bucket, 'statsig_example')
```
Comment (lfades) [inline edge-functions/ab-testing-statsig/pages/[bucket].tsx:55]: The same `UID_COOKIE` constant should be used here

**Pull request merged**